Neos.ContentRepository.Core/Classes/EventStore/EventPersister.php

```diff
@@ -23,18 +28,18 @@
 {
     public function __construct(
         private EventStoreInterface $eventStore,
-        private ProjectionCatchUpTriggerInterface $projectionCatchUpTrigger,
         private EventNormalizer $eventNormalizer,
-        private Projections $projections,
+        private ProjectionsAndCatchUpHooks $projectionsAndCatchUpHooks
     ) {
     }
 
     /**
      * @param EventsToPublish $eventsToPublish
+     * @param ContentRepository $contentRepository only required to build the possible catchup hooks
      * @return CommandResult
      * @throws ConcurrencyException in case the expectedVersion does not match
      */
-    public function publishEvents(EventsToPublish $eventsToPublish): CommandResult
+    public function publishEvents(EventsToPublish $eventsToPublish, ContentRepository $contentRepository): CommandResult
```

Comment:
having to pass the full cr along here seems wrong but is the only way if the EventPersister wants to be a separate class and not inlined in the ContentRepository

Previously the dependency problem was not really visible as we used the `contentRepositoryRegistry` to get the content repository in a `ProjectionCatchUpTriggerInterface` but there was no dependency flow set up then as well.